addons/hr_expense/models/hr_expense.py

```diff
@@ -0,0 +1,320 @@
+    analytic_account_id = fields.Many2one('account.analytic.account', string='Contract', states={'post': [('readonly', True)], 'done': [('readonly', True)]})
+    department_id = fields.Many2one('hr.department', string='Department', states={'post': [('readonly', True)], 'done': [('readonly', True)]})
+    description = fields.Text()
+    payment_mode = fields.Selection([('own_account', 'By Employee'), ('company_account', 'With Company Card')], default='own_account', states={'done': [('readonly', True)], 'post': [('readonly', True)]})
+    journal_id = fields.Many2one('account.journal', string='Force Journal', states={'done': [('readonly', True)], 'post': [('readonly', True)]}, domain=[('type', '=', ['case', 'bank'])], help = "The journal used when the expense is done.")
+    account_move_id = fields.Many2one('account.move', string='Ledger Posting', copy=False, track_visibility="onchange")
+    attachment_number = fields.Integer(compute='_compute_attachment_number', string='Number of Attachments')
+    reason = fields.Char(string="Reason", track_visibility="always")
+    state = fields.Selection([('draft', 'To Submit'),
+                              ('submit', 'Submitted'),
+                              ('approve', 'Approved'),
+                              ('post', 'Waiting Payment'),
+                              ('done', 'Paid'),
+                              ('cancel', 'Refused')
+                              ], string='Status', readonly=True, track_visibility='onchange', copy=False, default='draft', required=True,
+        help='When the expense request is created the status is \'To Submit\'.\n It is submitted by the employee and request is sent to manager, the status is \'Submitted\'.\
+        \nIf the manager approve it, the status is \'Approved\'.\n If the accountant genrate the accounting entries for the expense request, the status is \'Waiting Payment\'.')
+
+    @api.depends('quantity', 'unit_amount', 'tax_ids', 'currency_id')
+    def _compute_amount(self):
+        for expense in self:
+            expense.untaxed_amount = expense.unit_amount * expense.quantity
+            taxes = expense.tax_ids.compute_all(expense.unit_amount, expense.currency_id, expense.quantity, expense.product_id, expense.employee_id.user_id.partner_id)
+            expense.total_amount = taxes.get('total_included', False)
+
+    @api.multi
+    def _compute_attachment_number(self):
+        self.attachment_number = self.env['ir.attachment'].search_count([('res_model', '=', 'hr.expense'), ('res_id', 'in', self.ids)])
+
+    @api.onchange('product_id')
+    def _onchange_product_id(self):
+        if self.product_id:
+            self.unit_amount = self.env['product.template']._price_get(self.product_id, 'standard_price')[self.product_id.id]
+            self.product_uom_id = self.product_id.uom_id.id
```

Comment:
"self.product_uom_id = self.product_id.uom_id" is enough(could you please do the similar change wherever applicable in this branch)